source/common/http/utility.h

```diff
@@ -243,10 +243,30 @@ absl::string_view findQueryStringStart(const HeaderString& path);
 /**
  * Parse a particular value out of a cookie
  * @param headers supplies the headers to get the cookie from.
- * @param key the key for the particular cookie value to return
- * @return std::string the parsed cookie value, or "" if none exists
+ * @param key the key for the particular cookie value to return.
+ * @return absl::string_view the parsed cookie value, or a default constructed absl::string_view if
+ *         it doesn't exist.
  **/
-std::string parseCookieValue(const HeaderMap& headers, const std::string& key);
+absl::string_view parseCookieValue(const HeaderMap& headers, const absl::string_view key);
```

Comment:
This really just gives you the last key=value pair, right? I think we should rename the function to make it clear, i.e. `findLastOfCookieValue`

Reply:
This is indeed what it does. I didn't change this semantic (it was already doing this), only `s/std::string/absl::string_view/` and made sure not to duplicate code with  `parseCookieValues`.

I'm okay with remaining it if it fits with this PR :)

Reply:
Oh, yes, I think this is a very good point. There are obviously no tests that tested this quirky behavior. To comply with our change policy we should flag protect this change and allow users to revert to the old (arguably buggy) behavior, even though my sense is that there will be 0 users who will be impacted. 

@alyssawilk for opinion

Reply:
If it's legal to do this (cc @DavidSchinazi) then yeah, as long as the behavior is "more correct" we runtime guard, doc it in release notes, and land :-)

Reply:
From [the spec](https://datatracker.ietf.org/doc/html/rfc6265#section-5.4):
> When the user agent generates an HTTP request, the user agent MUST NOT attach more than one Cookie header field.

Because of this, I think taking the last `Cookie:` header and ignoring any other ones sounds reasonable to me.

However, [the spec also states](https://datatracker.ietf.org/doc/html/rfc6265#section-4.2.2):

> Although cookies are serialized linearly in the Cookie header, servers SHOULD NOT rely upon the serialization order.  In particular, if the Cookie header contains two cookies with the same name (e.g., that were set with different Path or Domain attributes), servers SHOULD NOT rely upon the order in which these cookies appear in the header.

So it's legal for a client to send `Cookie: foo=b;foo=c`.

This means that the original behavior before this PR could somewhat violate the spec. All that said, in practice this condition would be self-inflicted by the server which had sent two `Set-Cookie` headers with the same cookie-name and a different cookie-value. So as long as we don't do that we're fine either way?

Reply:
>  All that said, in practice this condition would be self-inflicted by the server which had sent two Set-Cookie headers with the same cookie-name and a different cookie-value. So as long as we don't do that we're fine either way?

I was in the perspective of having a "malicious" client that sends that kind of headers, for instance to try and bypass a matching filter that will be based on one choice for the `foo` cookie (let's say the first one), where the end server would use another value (e.g. the last one). But your scenario is also something to consider!

Note that, in the case of this iput matcher, we would emit all the possible cookie values separated by a comma. This should mimic the behavior of the header input matcher.

> we runtime guard, doc it in release notes

@alyssawilk is there an example of the kind of runtime guard I should add? Thanks for any info :)

Reply:
https://github.com/envoyproxy/envoy/blob/main/CONTRIBUTING.md#runtime-guarding

has general instructions and you can grep for reloadable in the code base.  If that's not enough to get you going lmk

Reply:
Thanks @alyssawilk for the hint! I should be able to manage from there.

Reply:
Implemented in https://github.com/envoyproxy/envoy/pull/16344/commits/828e6a112ec78fc8cec66fe67251d278db0e1d23 . Let me know @alyssawilk if I missed some parts! Thanks :)

Reply:
haha, checked back in since everyone forgets to test old behavior and you nailed it!  I'll let Yan resume the review but thanks for doing this step :-)